lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -177,6 +177,11 @@ TypeSystemSwiftTypeRef::GetBaseName(swift::Demangle::NodePointer node) {
   }
 }
 
+bool TypeSystemSwiftTypeRef::IsKnownSpecialImportedType(llvm::StringRef name) {
+  return name == "NSNotificationName" ||
```

Comment:
the lone hard coded NSNotificationName seems like a red flag. Is there really only one non-CF type at this time? What about future types.

Reply:
NSNotificationName really is special cased in the compiler 😅  https://github.com/apple/swift/blob/main/lib/ClangImporter/ClangAdapter.cpp#L553

Pointers, in the general case, are exposed in swift as `Unsafe...Pointer`, so don't go through the external provider, which is  I believe the only pointer types we should be queried about are ones that have special compiler handling. I put in the `lldb_assert` in case I'm wrong, and there are more types we should worry about.